scuba/config.py

```diff
@@ -261,6 +261,7 @@ def _get_volumes(data):
 
     vols = {}
     for cpath, v in voldata.items():
+        cpath = os.path.expandvars(cpath)
```

Comment:
It's unfortunate that [`os.path.expandvars()`](https://docs.python.org/3/library/os.path.html#os.path.expandvars) is so lenient and doesn't provide a `strict` option:

> Malformed variable names and references to non-existing variables are left unchanged.

How do you feel about using this [alternative](https://stackoverflow.com/a/37800939/119527) based on [`string.Template`](https://docs.python.org/3/library/string.html#template-strings)?

```python
def expandvars(s):
    return string.Template(s).substitute(os.environ)
```

If we use this, we'll need to handle the `KeyError` that can be raised if a variable is not set. Perhaps we map it to a `ConfigError`?

I would like to defer this to a future enhancement, but it's far easier to make a program less strict in a future release than it is to make it more strict.

Either way, I like having the indirection of a single function where expansion (of any kind) is handled, rather than multiple calls to `os.path.expandvars()`. Perhaps `path_expand`?

Reply:
Changed in 8a8c680. The code now uses `string.Template.substitute`, and provides both a general purpose expansion function in `utils.py` and a config-specific implementation (that raises the right exception) in `config.py`.